### ISSUE TITLE
Wrong SAGE since switching over from xdrip to Dexcom patched app

### ISSUE THREAD

**Issue by CaptainBalou** [MilosKozak/AndroidAPS#2140]
AndroidAPS
Version: 2.4-dev-h
Build: 9ffed9943-2019.10.10-13:48

Nightscout
Version: 0.12.6-dev

Dexcom patched app:
Version: com.dexcom.g6.region3.mgdl-patched-20191020.apk

xdrip (previously in use to read Dexcom G6 values)
Version: always latest nightly

**Behaviour:**
The sensor age in CP (care portal) and on the main screen is a huge negative number (see screenshot). This happens since I moved from xdrip to Dexcom patched app. While using xdrip all numbers where fine. The number have been reset to 0d after issueing "insert new sensor" in CP. Parallel to changing the app to get the bg values I did not changed the version of Nightscout.

![20191026_132028](https://user-images.githubusercontent.com/17015872/67620502-2377ee80-f808-11e9-8b83-1260d79ec963.jpg)

I also see that there is no value seen in Nightscout (see screenshot).

![Ohne Titel](https://user-images.githubusercontent.com/17015872/67620692-dd238f00-f809-11e9-8565-aa401c00b165.png)

Current AndroidAPS settings for Dexcom patched app.

![Screenshot_20191026-160221_AndroidAPS](https://user-images.githubusercontent.com/17015872/67620727-34296400-f80a-11e9-8aaa-2514338e5679.jpg)

**Procedure:**

1. Starting the new sensor via Dexcom patched app.
2. Issueing "insert new sensor" in CP (normally after that the number was 0)
3. Waiting 2h
4. Getting the first value which is perfectly transferred to AndroidAPS

**Expected behaviour:**
The expected number would be the number of days since "insert new sensor" in CP.

--------------------------------------------------

Can anyone think of what is the issue here? If needed I can provide more details whenever someone asks for it.

Thanks in advance

Marco

**Comment by ShaggydogUK** [comment]
For some reason APS adds in a treatment way in the future when you do an sensor change.  You can go into treatments, careportal, and delete the first thing there and the time will be back as it should be, as the correct time is also entered.  This is a workaround, not a solution!

**Comment by CaptainBalou** [comment]
Thanks for this hint @ShaggydogUK . It was exactly as you mentioned.

![20191027_205058](https://user-images.githubusercontent.com/17015872/67640694-62876c00-f8fc-11e9-931d-aef17b0bb79d.jpg)

I then removed this treatment. After this the time the sensor is already running is displayed correctly.

![Screenshot_20191027-205142_AndroidAPS](https://user-images.githubusercontent.com/17015872/67640713-864ab200-f8fc-11e9-906f-21e1f46648b8.jpg)

Does anyone know if this "old treatment in the future" is from AndroidAPS or from Nightscout? If this issue is not from AndroidAPS I would close this and re-open an issue in the Nightscout project tracker.

**Comment by CaptainBalou** [comment]
I put a new G6 sensor in place and started this one just by using the patched Dexcom app. I then realized that I had the same behaviour and got this old treatment in the future.

![20191030_230537](https://user-images.githubusercontent.com/17015872/67904278-2a875f80-fb6e-11e9-8c4e-65b1edb0a607.jpg)

So this errneous event may be generated by the patched app or is being generated by AndroidAPS after a trigger event of Dexcom patched app.

I then deleted the errneous entry and after that everything was displayed correct.
